SDL_gpu_shadercross.h

```diff
@@ -38,23 +38,430 @@ SDL_GpuShaderFormat SDL_ShaderCross_GetShaderFormats(void)
       | SDL_GPU_SHADERFORMAT_MSL
 #endif
 #if SDL_GPU_SHADERCROSS_HLSL
+      | SDL_GPU_SHADERFORMAT_SPIRV
       | SDL_GPU_SHADERFORMAT_DXBC
-      /* TODO: | SDL_GPU_SHADERFORMAT_DXIL */
+      | SDL_GPU_SHADERFORMAT_DXIL
 #endif /* SDL_GPU_SHADERCROSS_HLSL */
     );
 }
 
 #if SDL_GPU_SHADERCROSS_HLSL
 
 /* Win32 Type Definitions */
+
 typedef int HRESULT;
 typedef const void *LPCVOID;
 typedef size_t SIZE_T;
 typedef const char *LPCSTR;
 typedef unsigned int UINT;
 typedef unsigned long ULONG;
 typedef void *LPVOID;
-typedef void *REFIID;          /* hack, unused */
+typedef void *REFIID;
+
+/* DXIL via DXC */
+
+/* dxcompiler Type Definitions */
+typedef int BOOL;
+typedef void *REFCLSID;
+typedef wchar_t *LPCWSTR;
+
+/* Dynamic Library / Linking */
+#ifdef DXCOMPILER_DLL
+#undef DXCOMPILER_DLL
+#endif
+#if defined(_WIN32)
+#define DXCOMPILER_DLL "dxcompiler.dll"
+#elif defined(__APPLE__)
+#define DXCOMPILER_DLL "libdxcompiler.dylib"
+#else
+#define DXCOMPILER_DLL "libdxcompiler.so"
+#endif
+
+/* Unlike vkd3d-utils, libdxcompiler.so does not use msabi */
+#if !defined(_WIN32)
+#define __stdcall
+#endif
+
+/* Compiler Interface, _technically_ unofficial but it's MS C++, come on */
+typedef enum DXC_OUT_KIND
+{
+    DXC_OUT_NONE = 0,
+    DXC_OUT_OBJECT = 1,
+    DXC_OUT_ERRORS = 2,
+    DXC_OUT_PDB = 3,
+    DXC_OUT_SHADER_HASH = 4,
+    DXC_OUT_DISASSEMBLY = 5,
+    DXC_OUT_HLSL = 6,
+    DXC_OUT_TEXT = 7,
+    DXC_OUT_REFLECTION = 8,
+    DXC_OUT_ROOT_SIGNATURE = 9,
+    DXC_OUT_EXTRA_OUTPUTS = 10,
+    DXC_OUT_REMARKS = 11,
+    DXC_OUT_TIME_REPORT = 12,
+    DXC_OUT_TIME_TRACE = 13,
+    DXC_OUT_LAST = DXC_OUT_TIME_TRACE,
+    DXC_OUT_NUM_ENUMS,
+    DXC_OUT_FORCE_DWORD = 0xFFFFFFFF
+} DXC_OUT_KIND;
+
+typedef struct DxcBuffer
+{
+    LPCVOID Ptr;
+    SIZE_T Size;
+    UINT Encoding;
+} DxcBuffer;
+
+static Uint8 IID_IDxcBlob[] = {
+    0x08, 0xFB, 0xA5, 0x8B,
+    0x95, 0x51,
+    0xE2, 0x40,
+    0xAC,
+    0x58,
+    0x0D,
+    0x98,
+    0x9C,
+    0x3A,
+    0x01,
+    0x02
+};
+typedef struct IDxcBlob IDxcBlob;
+typedef struct IDxcBlobVtbl
+{
+    HRESULT(__stdcall *QueryInterface)(IDxcBlob *This, REFIID riid, void **ppvObject);
+    ULONG(__stdcall *AddRef)(IDxcBlob *This);
+    ULONG(__stdcall *Release)(IDxcBlob *This);
+
+    LPVOID(__stdcall *GetBufferPointer)(IDxcBlob *This);
+    SIZE_T(__stdcall *GetBufferSize)(IDxcBlob *This);
```

Comment:
Was really confused by this formatting for a moment. The lack of space between the return type and `(` makes it read like a function-like macro call.